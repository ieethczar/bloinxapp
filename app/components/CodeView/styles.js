import { StyleSheet } from "react-native";

import * as theme from '../../theme';

const styles = StyleSheet.create({
  text: {
    textAlign: 'center',
    color: theme.grayText,
    fontSize: 18,
  },
  codeView: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'center',
    marginBottom: 20,
    marginTop: 20,
  },
  digit: {
    backgroundColor: theme.selectionColor,
    margin: 3,
  }
});

export default styles;