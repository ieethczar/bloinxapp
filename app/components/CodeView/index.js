import React from 'react';
import PropTypes from 'prop-types';
import { View } from 'react-native';
import { Text, Avatar } from 'react-native-elements';

import styles from './styles';

const avatarOptions = {
  rounded: true,
  size: "medium",
  containerStyle: styles.digit,
};

const CodeView = ({ code }) => {
  const codeSplit = code ? code.split(''): '';

  if(!code || code.length !== 6) {
    new Error('El componente espera una cadena de 6 digitos.');
    return (
      <View style={styles.container}>
        <Text style={styles.text}>Código invalido</Text>
      </View>
    );
  }

  return (
    <View style={styles.container}>
      <Text style={styles.text}>Código de tanda</Text>
      <View style={styles.codeView}>
        <Avatar
          title={codeSplit[0]}
          {...avatarOptions}
        />
        <Avatar
          title={codeSplit[1]}
          {...avatarOptions}
        />
        <Avatar
          title={codeSplit[2]}
          {...avatarOptions}
        />
        <Avatar
          title={codeSplit[3]}
          {...avatarOptions}
        />
        <Avatar
          title={codeSplit[4]}
          {...avatarOptions}
        />
        <Avatar
          title={codeSplit[5]}
          {...avatarOptions}
        />
      </View>
    </View>
  );
}

CodeView.propTypes = {
  code: PropTypes.string,
};

export default CodeView;
