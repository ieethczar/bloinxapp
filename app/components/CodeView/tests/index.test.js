import React from 'react';
import renderer from 'react-test-renderer';

import CodeView from '../index';

describe('UserGlobe', () => {
  it('should render correctly', () => {
    const component = renderer.create(
      <CodeView />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly', () => {
    const component = renderer.create(
      <CodeView code="123456" />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});