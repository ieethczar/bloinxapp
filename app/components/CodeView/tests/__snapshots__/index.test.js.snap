// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserGlobe should render correctly 1`] = `
<View>
  <Text
    style={
      Object {
        "color": "#555555",
        "fontSize": 18,
        "textAlign": "center",
      }
    }
  >
    Código invalido
  </Text>
</View>
`;

exports[`UserGlobe should render correctly 2`] = `
<View>
  <Text
    style={
      Object {
        "color": "#555555",
        "fontSize": 18,
        "textAlign": "center",
      }
    }
  >
    Código de tanda
  </Text>
  <View
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "center",
        "marginBottom": 20,
        "marginTop": 20,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#407BFF",
          "borderRadius": 25,
          "height": 50,
          "margin": 3,
          "width": 50,
        }
      }
      theme={
        Object {
          "colors": Object {
            "black": "#242424",
            "disabled": "hsl(208, 8%, 90%)",
            "divider": "#bcbbc1",
            "error": "#ff190c",
            "grey0": "#393e42",
            "grey1": "#43484d",
            "grey2": "#5e6977",
            "grey3": "#86939e",
            "grey4": "#bdc6cf",
            "grey5": "#e1e8ee",
            "greyOutline": "#bbb",
            "platform": Object {
              "android": Object {
                "error": "#f44336",
                "grey": "rgba(0, 0, 0, 0.54)",
                "primary": "#2196f3",
                "searchBg": "#dcdce1",
                "secondary": "#9C27B0",
                "success": "#4caf50",
                "warning": "#ffeb3b",
              },
              "default": Object {
                "error": "#ff3b30",
                "grey": "#7d7d7d",
                "primary": "#007aff",
                "searchBg": "#dcdce1",
                "secondary": "#5856d6",
                "success": "#4cd964",
                "warning": "#ffcc00",
              },
              "ios": Object {
                "error": "#ff3b30",
                "grey": "#7d7d7d",
                "primary": "#007aff",
                "searchBg": "#dcdce1",
                "secondary": "#5856d6",
                "success": "#4cd964",
                "warning": "#ffcc00",
              },
              "web": Object {
                "error": "#ff190c",
                "grey": "#393e42",
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#ca71eb",
                "success": "#52c41a",
                "warning": "#faad14",
              },
            },
            "primary": "#2089dc",
            "searchBg": "#303337",
            "secondary": "#ca71eb",
            "success": "#52c41a",
            "warning": "#faad14",
            "white": "#ffffff",
          },
        }
      }
    >
      <View
        accessibilityIgnoresInvertColors={true}
        style={
          Object {
            "backgroundColor": "transparent",
            "borderRadius": 25,
            "flex": 1,
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <Image
          borderRadius={25}
          onLoad={[Function]}
          style={
            Object {
              "bottom": 0,
              "flex": 1,
              "height": undefined,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": undefined,
            }
          }
          testID="RNE__Image"
          theme={
            Object {
              "colors": Object {
                "black": "#242424",
                "disabled": "hsl(208, 8%, 90%)",
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "platform": Object {
                  "android": Object {
                    "error": "#f44336",
                    "grey": "rgba(0, 0, 0, 0.54)",
                    "primary": "#2196f3",
                    "searchBg": "#dcdce1",
                    "secondary": "#9C27B0",
                    "success": "#4caf50",
                    "warning": "#ffeb3b",
                  },
                  "default": Object {
                    "error": "#ff3b30",
                    "grey": "#7d7d7d",
                    "primary": "#007aff",
                    "searchBg": "#dcdce1",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                  "ios": Object {
                    "error": "#ff3b30",
                    "grey": "#7d7d7d",
                    "primary": "#007aff",
                    "searchBg": "#dcdce1",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                  "web": Object {
                    "error": "#ff190c",
                    "grey": "#393e42",
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#ca71eb",
                    "success": "#52c41a",
                    "warning": "#faad14",
                  },
                },
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#ca71eb",
                "success": "#52c41a",
                "warning": "#faad14",
                "white": "#ffffff",
              },
            }
          }
          transition={true}
          transitionDuration={360}
        />
        <View
          accessibilityElementsHidden={false}
          importantForAccessibility="yes"
          pointerEvents="auto"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "flex": 1,
                "height": undefined,
                "justifyContent": "center",
                "width": undefined,
              }
            }
            testID="RNE__Image__placeholder"
          >
            <Text
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "#ffffff",
                  "fontSize": 25,
                  "textAlign": "center",
                }
              }
            >
              1
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "height": undefined,
              "width": undefined,
            }
          }
          testID="RNE__Image__children__container"
        />
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#407BFF",
          "borderRadius": 25,
          "height": 50,
          "margin": 3,
          "width": 50,
        }
      }
      theme={
        Object {
          "colors": Object {
            "black": "#242424",
            "disabled": "hsl(208, 8%, 90%)",
            "divider": "#bcbbc1",
            "error": "#ff190c",
            "grey0": "#393e42",
            "grey1": "#43484d",
            "grey2": "#5e6977",
            "grey3": "#86939e",
            "grey4": "#bdc6cf",
            "grey5": "#e1e8ee",
            "greyOutline": "#bbb",
            "platform": Object {
              "android": Object {
                "error": "#f44336",
                "grey": "rgba(0, 0, 0, 0.54)",
                "primary": "#2196f3",
                "searchBg": "#dcdce1",
                "secondary": "#9C27B0",
                "success": "#4caf50",
                "warning": "#ffeb3b",
              },
              "default": Object {
                "error": "#ff3b30",
                "grey": "#7d7d7d",
                "primary": "#007aff",
                "searchBg": "#dcdce1",
                "secondary": "#5856d6",
                "success": "#4cd964",
                "warning": "#ffcc00",
              },
              "ios": Object {
                "error": "#ff3b30",
                "grey": "#7d7d7d",
                "primary": "#007aff",
                "searchBg": "#dcdce1",
                "secondary": "#5856d6",
                "success": "#4cd964",
                "warning": "#ffcc00",
              },
              "web": Object {
                "error": "#ff190c",
                "grey": "#393e42",
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#ca71eb",
                "success": "#52c41a",
                "warning": "#faad14",
              },
            },
            "primary": "#2089dc",
            "searchBg": "#303337",
            "secondary": "#ca71eb",
            "success": "#52c41a",
            "warning": "#faad14",
            "white": "#ffffff",
          },
        }
      }
    >
      <View
        accessibilityIgnoresInvertColors={true}
        style={
          Object {
            "backgroundColor": "transparent",
            "borderRadius": 25,
            "flex": 1,
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <Image
          borderRadius={25}
          onLoad={[Function]}
          style={
            Object {
              "bottom": 0,
              "flex": 1,
              "height": undefined,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": undefined,
            }
          }
          testID="RNE__Image"
          theme={
            Object {
              "colors": Object {
                "black": "#242424",
                "disabled": "hsl(208, 8%, 90%)",
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "platform": Object {
                  "android": Object {
                    "error": "#f44336",
                    "grey": "rgba(0, 0, 0, 0.54)",
                    "primary": "#2196f3",
                    "searchBg": "#dcdce1",
                    "secondary": "#9C27B0",
                    "success": "#4caf50",
                    "warning": "#ffeb3b",
                  },
                  "default": Object {
                    "error": "#ff3b30",
                    "grey": "#7d7d7d",
                    "primary": "#007aff",
                    "searchBg": "#dcdce1",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                  "ios": Object {
                    "error": "#ff3b30",
                    "grey": "#7d7d7d",
                    "primary": "#007aff",
                    "searchBg": "#dcdce1",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                  "web": Object {
                    "error": "#ff190c",
                    "grey": "#393e42",
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#ca71eb",
                    "success": "#52c41a",
                    "warning": "#faad14",
                  },
                },
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#ca71eb",
                "success": "#52c41a",
                "warning": "#faad14",
                "white": "#ffffff",
              },
            }
          }
          transition={true}
          transitionDuration={360}
        />
        <View
          accessibilityElementsHidden={false}
          importantForAccessibility="yes"
          pointerEvents="auto"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "flex": 1,
                "height": undefined,
                "justifyContent": "center",
                "width": undefined,
              }
            }
            testID="RNE__Image__placeholder"
          >
            <Text
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "#ffffff",
                  "fontSize": 25,
                  "textAlign": "center",
                }
              }
            >
              2
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "height": undefined,
              "width": undefined,
            }
          }
          testID="RNE__Image__children__container"
        />
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#407BFF",
          "borderRadius": 25,
          "height": 50,
          "margin": 3,
          "width": 50,
        }
      }
      theme={
        Object {
          "colors": Object {
            "black": "#242424",
            "disabled": "hsl(208, 8%, 90%)",
            "divider": "#bcbbc1",
            "error": "#ff190c",
            "grey0": "#393e42",
            "grey1": "#43484d",
            "grey2": "#5e6977",
            "grey3": "#86939e",
            "grey4": "#bdc6cf",
            "grey5": "#e1e8ee",
            "greyOutline": "#bbb",
            "platform": Object {
              "android": Object {
                "error": "#f44336",
                "grey": "rgba(0, 0, 0, 0.54)",
                "primary": "#2196f3",
                "searchBg": "#dcdce1",
                "secondary": "#9C27B0",
                "success": "#4caf50",
                "warning": "#ffeb3b",
              },
              "default": Object {
                "error": "#ff3b30",
                "grey": "#7d7d7d",
                "primary": "#007aff",
                "searchBg": "#dcdce1",
                "secondary": "#5856d6",
                "success": "#4cd964",
                "warning": "#ffcc00",
              },
              "ios": Object {
                "error": "#ff3b30",
                "grey": "#7d7d7d",
                "primary": "#007aff",
                "searchBg": "#dcdce1",
                "secondary": "#5856d6",
                "success": "#4cd964",
                "warning": "#ffcc00",
              },
              "web": Object {
                "error": "#ff190c",
                "grey": "#393e42",
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#ca71eb",
                "success": "#52c41a",
                "warning": "#faad14",
              },
            },
            "primary": "#2089dc",
            "searchBg": "#303337",
            "secondary": "#ca71eb",
            "success": "#52c41a",
            "warning": "#faad14",
            "white": "#ffffff",
          },
        }
      }
    >
      <View
        accessibilityIgnoresInvertColors={true}
        style={
          Object {
            "backgroundColor": "transparent",
            "borderRadius": 25,
            "flex": 1,
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <Image
          borderRadius={25}
          onLoad={[Function]}
          style={
            Object {
              "bottom": 0,
              "flex": 1,
              "height": undefined,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": undefined,
            }
          }
          testID="RNE__Image"
          theme={
            Object {
              "colors": Object {
                "black": "#242424",
                "disabled": "hsl(208, 8%, 90%)",
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "platform": Object {
                  "android": Object {
                    "error": "#f44336",
                    "grey": "rgba(0, 0, 0, 0.54)",
                    "primary": "#2196f3",
                    "searchBg": "#dcdce1",
                    "secondary": "#9C27B0",
                    "success": "#4caf50",
                    "warning": "#ffeb3b",
                  },
                  "default": Object {
                    "error": "#ff3b30",
                    "grey": "#7d7d7d",
                    "primary": "#007aff",
                    "searchBg": "#dcdce1",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                  "ios": Object {
                    "error": "#ff3b30",
                    "grey": "#7d7d7d",
                    "primary": "#007aff",
                    "searchBg": "#dcdce1",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                  "web": Object {
                    "error": "#ff190c",
                    "grey": "#393e42",
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#ca71eb",
                    "success": "#52c41a",
                    "warning": "#faad14",
                  },
                },
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#ca71eb",
                "success": "#52c41a",
                "warning": "#faad14",
                "white": "#ffffff",
              },
            }
          }
          transition={true}
          transitionDuration={360}
        />
        <View
          accessibilityElementsHidden={false}
          importantForAccessibility="yes"
          pointerEvents="auto"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "flex": 1,
                "height": undefined,
                "justifyContent": "center",
                "width": undefined,
              }
            }
            testID="RNE__Image__placeholder"
          >
            <Text
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "#ffffff",
                  "fontSize": 25,
                  "textAlign": "center",
                }
              }
            >
              3
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "height": undefined,
              "width": undefined,
            }
          }
          testID="RNE__Image__children__container"
        />
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#407BFF",
          "borderRadius": 25,
          "height": 50,
          "margin": 3,
          "width": 50,
        }
      }
      theme={
        Object {
          "colors": Object {
            "black": "#242424",
            "disabled": "hsl(208, 8%, 90%)",
            "divider": "#bcbbc1",
            "error": "#ff190c",
            "grey0": "#393e42",
            "grey1": "#43484d",
            "grey2": "#5e6977",
            "grey3": "#86939e",
            "grey4": "#bdc6cf",
            "grey5": "#e1e8ee",
            "greyOutline": "#bbb",
            "platform": Object {
              "android": Object {
                "error": "#f44336",
                "grey": "rgba(0, 0, 0, 0.54)",
                "primary": "#2196f3",
                "searchBg": "#dcdce1",
                "secondary": "#9C27B0",
                "success": "#4caf50",
                "warning": "#ffeb3b",
              },
              "default": Object {
                "error": "#ff3b30",
                "grey": "#7d7d7d",
                "primary": "#007aff",
                "searchBg": "#dcdce1",
                "secondary": "#5856d6",
                "success": "#4cd964",
                "warning": "#ffcc00",
              },
              "ios": Object {
                "error": "#ff3b30",
                "grey": "#7d7d7d",
                "primary": "#007aff",
                "searchBg": "#dcdce1",
                "secondary": "#5856d6",
                "success": "#4cd964",
                "warning": "#ffcc00",
              },
              "web": Object {
                "error": "#ff190c",
                "grey": "#393e42",
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#ca71eb",
                "success": "#52c41a",
                "warning": "#faad14",
              },
            },
            "primary": "#2089dc",
            "searchBg": "#303337",
            "secondary": "#ca71eb",
            "success": "#52c41a",
            "warning": "#faad14",
            "white": "#ffffff",
          },
        }
      }
    >
      <View
        accessibilityIgnoresInvertColors={true}
        style={
          Object {
            "backgroundColor": "transparent",
            "borderRadius": 25,
            "flex": 1,
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <Image
          borderRadius={25}
          onLoad={[Function]}
          style={
            Object {
              "bottom": 0,
              "flex": 1,
              "height": undefined,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": undefined,
            }
          }
          testID="RNE__Image"
          theme={
            Object {
              "colors": Object {
                "black": "#242424",
                "disabled": "hsl(208, 8%, 90%)",
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "platform": Object {
                  "android": Object {
                    "error": "#f44336",
                    "grey": "rgba(0, 0, 0, 0.54)",
                    "primary": "#2196f3",
                    "searchBg": "#dcdce1",
                    "secondary": "#9C27B0",
                    "success": "#4caf50",
                    "warning": "#ffeb3b",
                  },
                  "default": Object {
                    "error": "#ff3b30",
                    "grey": "#7d7d7d",
                    "primary": "#007aff",
                    "searchBg": "#dcdce1",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                  "ios": Object {
                    "error": "#ff3b30",
                    "grey": "#7d7d7d",
                    "primary": "#007aff",
                    "searchBg": "#dcdce1",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                  "web": Object {
                    "error": "#ff190c",
                    "grey": "#393e42",
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#ca71eb",
                    "success": "#52c41a",
                    "warning": "#faad14",
                  },
                },
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#ca71eb",
                "success": "#52c41a",
                "warning": "#faad14",
                "white": "#ffffff",
              },
            }
          }
          transition={true}
          transitionDuration={360}
        />
        <View
          accessibilityElementsHidden={false}
          importantForAccessibility="yes"
          pointerEvents="auto"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "flex": 1,
                "height": undefined,
                "justifyContent": "center",
                "width": undefined,
              }
            }
            testID="RNE__Image__placeholder"
          >
            <Text
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "#ffffff",
                  "fontSize": 25,
                  "textAlign": "center",
                }
              }
            >
              4
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "height": undefined,
              "width": undefined,
            }
          }
          testID="RNE__Image__children__container"
        />
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#407BFF",
          "borderRadius": 25,
          "height": 50,
          "margin": 3,
          "width": 50,
        }
      }
      theme={
        Object {
          "colors": Object {
            "black": "#242424",
            "disabled": "hsl(208, 8%, 90%)",
            "divider": "#bcbbc1",
            "error": "#ff190c",
            "grey0": "#393e42",
            "grey1": "#43484d",
            "grey2": "#5e6977",
            "grey3": "#86939e",
            "grey4": "#bdc6cf",
            "grey5": "#e1e8ee",
            "greyOutline": "#bbb",
            "platform": Object {
              "android": Object {
                "error": "#f44336",
                "grey": "rgba(0, 0, 0, 0.54)",
                "primary": "#2196f3",
                "searchBg": "#dcdce1",
                "secondary": "#9C27B0",
                "success": "#4caf50",
                "warning": "#ffeb3b",
              },
              "default": Object {
                "error": "#ff3b30",
                "grey": "#7d7d7d",
                "primary": "#007aff",
                "searchBg": "#dcdce1",
                "secondary": "#5856d6",
                "success": "#4cd964",
                "warning": "#ffcc00",
              },
              "ios": Object {
                "error": "#ff3b30",
                "grey": "#7d7d7d",
                "primary": "#007aff",
                "searchBg": "#dcdce1",
                "secondary": "#5856d6",
                "success": "#4cd964",
                "warning": "#ffcc00",
              },
              "web": Object {
                "error": "#ff190c",
                "grey": "#393e42",
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#ca71eb",
                "success": "#52c41a",
                "warning": "#faad14",
              },
            },
            "primary": "#2089dc",
            "searchBg": "#303337",
            "secondary": "#ca71eb",
            "success": "#52c41a",
            "warning": "#faad14",
            "white": "#ffffff",
          },
        }
      }
    >
      <View
        accessibilityIgnoresInvertColors={true}
        style={
          Object {
            "backgroundColor": "transparent",
            "borderRadius": 25,
            "flex": 1,
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <Image
          borderRadius={25}
          onLoad={[Function]}
          style={
            Object {
              "bottom": 0,
              "flex": 1,
              "height": undefined,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": undefined,
            }
          }
          testID="RNE__Image"
          theme={
            Object {
              "colors": Object {
                "black": "#242424",
                "disabled": "hsl(208, 8%, 90%)",
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "platform": Object {
                  "android": Object {
                    "error": "#f44336",
                    "grey": "rgba(0, 0, 0, 0.54)",
                    "primary": "#2196f3",
                    "searchBg": "#dcdce1",
                    "secondary": "#9C27B0",
                    "success": "#4caf50",
                    "warning": "#ffeb3b",
                  },
                  "default": Object {
                    "error": "#ff3b30",
                    "grey": "#7d7d7d",
                    "primary": "#007aff",
                    "searchBg": "#dcdce1",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                  "ios": Object {
                    "error": "#ff3b30",
                    "grey": "#7d7d7d",
                    "primary": "#007aff",
                    "searchBg": "#dcdce1",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                  "web": Object {
                    "error": "#ff190c",
                    "grey": "#393e42",
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#ca71eb",
                    "success": "#52c41a",
                    "warning": "#faad14",
                  },
                },
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#ca71eb",
                "success": "#52c41a",
                "warning": "#faad14",
                "white": "#ffffff",
              },
            }
          }
          transition={true}
          transitionDuration={360}
        />
        <View
          accessibilityElementsHidden={false}
          importantForAccessibility="yes"
          pointerEvents="auto"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "flex": 1,
                "height": undefined,
                "justifyContent": "center",
                "width": undefined,
              }
            }
            testID="RNE__Image__placeholder"
          >
            <Text
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "#ffffff",
                  "fontSize": 25,
                  "textAlign": "center",
                }
              }
            >
              5
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "height": undefined,
              "width": undefined,
            }
          }
          testID="RNE__Image__children__container"
        />
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#407BFF",
          "borderRadius": 25,
          "height": 50,
          "margin": 3,
          "width": 50,
        }
      }
      theme={
        Object {
          "colors": Object {
            "black": "#242424",
            "disabled": "hsl(208, 8%, 90%)",
            "divider": "#bcbbc1",
            "error": "#ff190c",
            "grey0": "#393e42",
            "grey1": "#43484d",
            "grey2": "#5e6977",
            "grey3": "#86939e",
            "grey4": "#bdc6cf",
            "grey5": "#e1e8ee",
            "greyOutline": "#bbb",
            "platform": Object {
              "android": Object {
                "error": "#f44336",
                "grey": "rgba(0, 0, 0, 0.54)",
                "primary": "#2196f3",
                "searchBg": "#dcdce1",
                "secondary": "#9C27B0",
                "success": "#4caf50",
                "warning": "#ffeb3b",
              },
              "default": Object {
                "error": "#ff3b30",
                "grey": "#7d7d7d",
                "primary": "#007aff",
                "searchBg": "#dcdce1",
                "secondary": "#5856d6",
                "success": "#4cd964",
                "warning": "#ffcc00",
              },
              "ios": Object {
                "error": "#ff3b30",
                "grey": "#7d7d7d",
                "primary": "#007aff",
                "searchBg": "#dcdce1",
                "secondary": "#5856d6",
                "success": "#4cd964",
                "warning": "#ffcc00",
              },
              "web": Object {
                "error": "#ff190c",
                "grey": "#393e42",
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#ca71eb",
                "success": "#52c41a",
                "warning": "#faad14",
              },
            },
            "primary": "#2089dc",
            "searchBg": "#303337",
            "secondary": "#ca71eb",
            "success": "#52c41a",
            "warning": "#faad14",
            "white": "#ffffff",
          },
        }
      }
    >
      <View
        accessibilityIgnoresInvertColors={true}
        style={
          Object {
            "backgroundColor": "transparent",
            "borderRadius": 25,
            "flex": 1,
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <Image
          borderRadius={25}
          onLoad={[Function]}
          style={
            Object {
              "bottom": 0,
              "flex": 1,
              "height": undefined,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": undefined,
            }
          }
          testID="RNE__Image"
          theme={
            Object {
              "colors": Object {
                "black": "#242424",
                "disabled": "hsl(208, 8%, 90%)",
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "platform": Object {
                  "android": Object {
                    "error": "#f44336",
                    "grey": "rgba(0, 0, 0, 0.54)",
                    "primary": "#2196f3",
                    "searchBg": "#dcdce1",
                    "secondary": "#9C27B0",
                    "success": "#4caf50",
                    "warning": "#ffeb3b",
                  },
                  "default": Object {
                    "error": "#ff3b30",
                    "grey": "#7d7d7d",
                    "primary": "#007aff",
                    "searchBg": "#dcdce1",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                  "ios": Object {
                    "error": "#ff3b30",
                    "grey": "#7d7d7d",
                    "primary": "#007aff",
                    "searchBg": "#dcdce1",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                  "web": Object {
                    "error": "#ff190c",
                    "grey": "#393e42",
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#ca71eb",
                    "success": "#52c41a",
                    "warning": "#faad14",
                  },
                },
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#ca71eb",
                "success": "#52c41a",
                "warning": "#faad14",
                "white": "#ffffff",
              },
            }
          }
          transition={true}
          transitionDuration={360}
        />
        <View
          accessibilityElementsHidden={false}
          importantForAccessibility="yes"
          pointerEvents="auto"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "flex": 1,
                "height": undefined,
                "justifyContent": "center",
                "width": undefined,
              }
            }
            testID="RNE__Image__placeholder"
          >
            <Text
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "#ffffff",
                  "fontSize": 25,
                  "textAlign": "center",
                }
              }
            >
              6
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "height": undefined,
              "width": undefined,
            }
          }
          testID="RNE__Image__children__container"
        />
      </View>
    </View>
  </View>
</View>
`;
